Library/OcAfterBootCompatLib/CustomSlide.c

```diff
@@ -89,11 +89,18 @@ GenerateSlideValue (
     return SlideSupport->ValidSlides[0];
   }
 
-  do {
-    DivU64x32Remainder (GetPseudoRandomNumber64 (), SlideSupport->ValidSlideCount, &Slide);
-  } while (Slide == 0);
+  //
+  // Handle maximum available slide case.
+  //
+  DivU64x32Remainder (
+    GetPseudoRandomNumber64 (),
+    SlideSupport->ProvideMaxSlide == 0
```

Comment:
Ugh, this is still wrong. In this implementation `SlideSupport->ProvideMaxSlide` will be maximum number of slides used. I.e. it says first N slides, but these first slides can be anything, they can be 1, 2, 3, 5, 6 or they can be 1, 2, 3, 5, 9, 100, 101, 102, 103 as `SlideSupport->ValidSlides` is a sparse list of values (!).

What I believe this should really do is to add a check [here](https://github.com/acidanthera/OpenCorePkg/blob/796765c/Library/OcAfterBootCompatLib/CustomSlide.c#L365), replacing it with something like:

```C
if (StartAddr + AvailableSize) != EndAddr
  || (SlideSupport->ProvideMaxSlide > 0 && Slide > SlideSupport->ProvideMaxSlide))
```

Reply:
Or if we can break the outer for-loop before that line and skip evaluating remaining higher slides?

Reply:
Hm, or actually that. Yeah, that's more obvious of course.